mysql/infoschema.go

```diff
@@ -0,0 +1,357 @@
+		if err != nil {
+			fmt.Printf("Can't scan: %v\n", err)
+			continue
+		}
+		unique := false
+		ignored := schema.Ignored{}
+		for _, c := range constraints[colName] {
+			// c can be UNIQUE, PRIMARY KEY, FOREIGN KEY or CHECK
+			// We've already filtered out PRIMARY KEY.
+			switch c {
+			case "UNIQUE":
+				unique = true
+			case "FOREIGN KEY":
+				ignored.ForeignKey = true
+			case "CHECK":
+				ignored.Check = true
+			}
+		}
+		ignored.Default = colDefault.Valid
+		c := schema.Column{
+			Name:    colName,
+			Type:    toType(dataType, columnType, charMaxLen, numericPrecision, numericScale),
+			NotNull: toNotNull(conv, isNullable),
+			Unique:  unique,
+			Ignored: ignored,
+		}
+		colDefs[colName] = c
+		colNames = append(colNames, colName)
+	}
+	return colDefs, colNames
+}
+
+// getConstraints returns a list of primary keys and by-column map of
+// other constraints.  Note: we need to preserve ordinal order of
+// columns in primary key constraints.
+func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]string, map[string][]string, error) {
+	q := `SELECT k.COLUMN_NAME, t.CONSTRAINT_TYPE
+              FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS t
+                INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS k
+                  ON t.CONSTRAINT_NAME = k.CONSTRAINT_NAME AND t.CONSTRAINT_SCHEMA = k.CONSTRAINT_SCHEMA AND t.TABLE_NAME=k.TABLE_NAME
+              WHERE k.TABLE_SCHEMA = ? AND k.TABLE_NAME = ? ORDER BY k.ordinal_position;`
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer rows.Close()
+	var primaryKeys []string
+	var col, constraint string
+	m := make(map[string][]string)
+	for rows.Next() {
+		err := rows.Scan(&col, &constraint)
+		if err != nil {
+			fmt.Printf("Can't scan: %v\n", err)
+			continue
+		}
+		if col == "" || constraint == "" {
+			fmt.Printf("Got empty col or constraint\n")
+			continue
+		}
+		switch constraint {
+		case "PRIMARY KEY":
+			primaryKeys = append(primaryKeys, col)
+		default:
+			m[col] = append(m[col], constraint)
+		}
+	}
+	return primaryKeys, m, nil
+}
+
+func toType(dataType string, columnType string, charLen sql.NullInt64, numericPrecision, numericScale sql.NullInt64) schema.Type {
+	switch {
+	case dataType == "set":
+		return schema.Type{Name: dataType, ArrayBounds: []int64{-1}}
+	case charLen.Valid && dataType != "enum":
+		return schema.Type{Name: dataType, Mods: []int64{charLen.Int64}}
+	case dataType == "decimal" && numericPrecision.Valid && numericScale.Valid && numericScale.Int64 != 0:
+		return schema.Type{Name: dataType, Mods: []int64{numericPrecision.Int64, numericScale.Int64}}
+	case dataType == "decimal" && numericPrecision.Valid:
+		return schema.Type{Name: dataType, Mods: []int64{numericPrecision.Int64}}
+	default:
+		return schema.Type{Name: dataType}
+	}
+}
+
+func toNotNull(conv *internal.Conv, isNullable string) bool {
+	switch isNullable {
+	case "YES":
```

Comment:
Are we sure if its going to be YES and no other cases

Reply:
yes, it can be only 'YES' or 'NO'.

Mentioned in mysql docs : ([LINK](https://dev.mysql.com/doc/mysql-infoschema-excerpt/5.6/en/columns-table.html))

- IS_NULLABLE

The column nullability. The value is YES if NULL values can be stored in the column, NO if not.